src/huggingface_hub/utils/_errors.py

```diff
@@ -1,250 +1,19 @@
-import re
 from typing import Optional
 
 from requests import HTTPError, Response
 
-from ._fixes import JSONDecodeError
-
-
-REPO_API_REGEX = re.compile(
-    r"""
-        # staging or production endpoint
-        ^https://[^/]+
-        (
-            # on /api/repo_type/repo_id
-            /api/(models|datasets|spaces)/(.+)
-            |
-            # or /repo_id/resolve/revision/...
-            /(.+)/resolve/(.+)
-        )
-    """,
-    flags=re.VERBOSE,
```

Comment:
This can be kept in `./utils/_errors.py` since it's not a Exception class definition